flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/bridging/BridgingSqlFunction.java

```diff
@@ -279,5 +279,15 @@ public SqlReturnTypeInference getRowTypeInference() {
             }
             return TableCharacteristic.builder(semantics).build();
         }
+
+        @Override
+        public boolean argumentMustBeScalar(int ordinal) {
+            final List<StaticArgument> args = typeInference.getStaticArguments().orElse(null);
+            if (args == null || ordinal >= args.size()) {
+                return true;
+            }
+            final StaticArgument arg = args.get(ordinal);
+            return !arg.is(StaticArgumentTrait.TABLE);
```

Comment:
should here be explicit check for scalar?
I'm asking since not sure that every non-`TABLE` fits here from
https://github.com/apache/flink/blob/1adcf1e47a7e6ae9908736c3969e2d279a44bc18/flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/StaticArgumentTrait.java#L35-L40

Reply:
> adding a new value could unintentionally break something in future
I'm already using a conservative approach here by not testing for for SCALAR. If you take a look at the default implementation in the super class, it is also `true` by default. So only TABLE args will set this to something else.

Reply:
makes sense
thanks for clarifying this